llvm/test/Transforms/GVNHoist/hoist-md.ll

```diff
@@ -157,6 +157,35 @@ return:                                           ; preds = %if.end, %if.then
   ret ptr %retval.0
 }
 
+; TODO: Should this work?
+define void @test6(i1 %b, ptr %x) {
```

Comment:
I'm not sure this is ok. I think we might want to inhibit the optimization in this case to avoid losing MMRAs too much.
GVN kicks in a lot and if we merge MMRAs a couple of times we can easily end up with the empty set again.